### PR TITLE
feat: add support for Azure CIAM login

### PR DESCRIPTION
**Pull request by hf**
Adds support for [Azure's CIAM](https://learn.microsoft.com/en-us/entra/external-id/customers/overview-customers-ciam) login. This is a special B2B Azure account separate from the typical tenant accounts and is meant to be used only when the expected issuer is set to the CIAM tenant.

### CODE REVIEW

File: internal/api/errorcodes.go [hunk 1]
```diff
@@ -47,7 +47,6 @@ const (
 	ErrorCodeSMSSendFailed                     ErrorCode = "sms_send_failed"
 	ErrorCodeEmailNotConfirmed                 ErrorCode = "email_not_confirmed"
 	ErrorCodePhoneNotConfirmed                 ErrorCode = "phone_not_confirmed"
-	ErrorCodeReauthNonceMissing                ErrorCode = "reauth_nonce_missing"
```
Comment (hf) [inline internal/api/errorcodes.go:50]: Found this error code which is not used any more.

**Review by kangmingtay — APPROVED** [review]

**Comment by kangmingtay** [comment]
@hf it doesn't seem trivial to test this one out :/ might be good to add some links to the docs to show what the issuer is expected to look like?

File: internal/api/provider/azure.go [hunk 2]
```diff
@@ -39,11 +39,16 @@ type azureProvider struct {
 }
 
 var azureIssuerRegexp = regexp.MustCompile("^https://login[.]microsoftonline[.]com/([^/]+)/v2[.]0/?$")
+var azureCIAMIssuerRegexp = regexp.MustCompile("^https://[a-z0-9-]+[.]ciamlogin[.]com/([^/]+)/v2[.]0/?$")
```
Comment (J0) [inline internal/api/provider/azure.go:42]: /nit 

It's an edge case but is there any restriction around tenant/subdomain naming?

It's rare but if the subdomain includes an underscore (e.g. `https://supabase_auth.ciamlogin.com/` it won't match

File: internal/api/provider/azure.go [hunk 3]
```diff
@@ -121,7 +126,10 @@ func (g azureProvider) GetUserData(ctx context.Context, tok *oauth2.Token) (*Use
 			return nil, err
 		}
 
-		if !IsAzureIssuer(issuer) {
+		// Allow basic Azure issuers, except when the expected issuer
+		// is configured to be the Azure CIAM issuer, allow CIAM
+		// issuers to pass.
+		if !IsAzureIssuer(issuer) || (IsAzureCIAMIssuer(g.ExpectedIssuer) && !IsAzureCIAMIssuer(issuer)) {
```
[the remark this answers is not in the document]
Comment (J0) [reply internal/api/provider/azure.go:132]: My interpretation of this statement that we'll reject cases where

1. We have non-azure issuers
2. We have expected CIAM issuers but the issuer is not a CIAM issuer

File: internal/api/provider/azure.go (hunk 2 above)
Comment (J0) [reply internal/api/provider/azure.go:42]: Seems like [letters and numbers only](https://techcommunity.microsoft.com/t5/microsoft-365/tenant-name-restrictions-in-o365/m-p/638184)

**Pull request merged**